src/daemons/runDaemon.ts

```diff
@@ -107,3 +107,24 @@ async function iteration(
   // Force-close any lingering stuff:
   process.exit(0)
 }
+
+/**
+ * Select a period of time to use for a loop iteration.
+ *
+ * For instance, if a transaction was sent in the past 10 minutes,
+ * we should check it every loop. If it has been in the mempool for a
+ * few weeks, it probably won't confirm any time soon,
+ * so we can check it every few days.
+ *
+ * If t is our timeout:
+ * - Check events younger than t on every loop
+ * - Check events younger than 2t on every other loop
+ * - Check events younger than 4t every 4 loops
+ * - Check events younger than 8t every 8 loops
+ * - etc...
+ */
+export function exponentialBackoff(iteration: number): number {
+  let lsb = 0
+  while ((iteration & (1 << lsb)) === 0 && lsb < 30) ++lsb
+  return 1 << lsb
+}
```

Comment:
Let's talk about this fn. I gather it will always return a number which is the factor by which to multiply the interval unit which used as a "look back time" in the database query. So I can see it starts out at 2^30,  and then increases from 1 to 2^30 in a jigsaw pattern.

Reply:
We had a chat about the "jigsaw pattern", and the bottom line is that it does revisit everything on a regular basis.

Every event will get checked at a "regular" frequency, with "regular" becoming slower and slower as the event ages. There is no case where we forget about an event and stop checking it altogether (aside from the confirmation timeout, which is a separate thing).